docs/source/notebooks/demo.ipynb

```diff
@@ -205,28 +314,84 @@
+    "matching_columns = [\"message\", \"date\", \"attachment\", \"attachement_type\", \"subject\"]\n",
+    "# the predefined keys that should be used to match these columns, in the correct order\n",
+    "pre_defined_keys = [\"content\", \"date\", \"attachment\", \"attachement_type\", \"subject\"]\n",
+    "# what to call any columns that are not matched to pre-defined keys\n",
+    "unmatched_keyword = \"unmatched\"\n",
+    "# or get the unmatched keyword from the workflow settings\n",
     "unmatched_keyword = workflow_settings.get(\"csv_col_unmatched_keyword\")\n",
+    "\n",
     "input_handler = mailcom.get_input_handler(in_path=input_csv, in_type=\"csv\", \n",
-    "                                          col_names=[\"message\"], \n",
-    "                                          init_data_fields=[\"content\"], \n",
+    "                                          col_names=matching_columns, \n",
+    "                                          init_data_fields=pre_defined_keys, \n",
     "                                          unmatched_keyword=unmatched_keyword)"
    ]
   },
   {
    "cell_type": "markdown",
    "metadata": {},
    "source": [
-    "In the cell above, the `message` column from the `csv` file is mapped to the `content` key in the email dictionary, while other keys have `None` as their values.\n",
+    "In the cell above, the `message` column from the `csv` file is mapped to the `content` key in the email dictionary. Any additional columns of the `csv` file are discarded."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "pp.pprint(input_handler.email_list[0])"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# path to your csv file - change this to your own file\n",
+    "input_csv = \"../../../data/in/sample_data.csv\"\n",
+    "# the columns of the csv that should be passed through the processing pipeline/retained in the pipeline\n",
+    "matching_columns = [\"message\", \"date\"]\n",
+    "# the predefined keys that should be used to match these columns, in the correct order\n",
+    "pre_defined_keys = [\"content\", \"date\"]\n",
+    "# what to call any columns that are not matched to pre-defined keys\n",
+    "unmatched_keyword = \"unmatched\"\n",
+    "# or get the unmatched keyword from the workflow settings\n",
+    "unmatched_keyword = workflow_settings.get(\"csv_col_unmatched_keyword\")\n",
     "\n",
-    "If the `csv` file lacks the `message` column, value of `content` is set to `unmatched_keyword`"
+    "input_handler = mailcom.get_input_handler(in_path=input_csv, in_type=\"csv\", \n",
+    "                                          col_names=matching_columns, \n",
+    "                                          init_data_fields=pre_defined_keys, \n",
+    "                                          unmatched_keyword=unmatched_keyword)"
```

Comment:
can we make sure this works as intended, or remove the keyword? It looks like unmatched columns are not retained, contrary to what I would have expected from the keyword description in the schema/docstrings.